### PR TITLE
APPS: Drop interactive mode in the 'openssl' program

### PR DESCRIPTION
**Pull request by levitte**
This mode is severely untested and unmaintained, is seems not to be
used very much.

### CODE REVIEW

**Comment by paulidale** [comment]
Since this is a breaking change, we'll need an OMC vote.

**Comment by t8m** [comment]
Maybe also add a RIP note in the HISTORY section of openssl(1) manpage?

**Comment by t8m** [comment]
The poor and unwanted openssl interactive application mode died without suffering in OpenSSL 3.0 version.

**Comment by levitte** [comment]
> Maybe also add a RIP note in the HISTORY section of openssl(1) manpage?

It turns out that interactive mode isn't even documented, not even in 1.0.2.

**Comment by levitte** [comment]
> not even in 1.0.2.

I went looking as far back as 0.9.8, still not documented.

**Comment by levitte** [comment]
note added in the HISTORY section in `doc/man1/openssl.pod`

File: doc/man1/openssl.pod [hunk 1]
```diff
@@ -1405,7 +1405,11 @@ The B<-issuer_checks> option is deprecated as of OpenSSL 1.1.0 and
 is silently ignored.
 
 The B<-xcertform> and B<-xkeyform> options
-are obsolete since OpenSSL 3.0.0 and have no effect.
+are obsolete since OpenSSL 3.0 and have no effect.
+
+The interactive mode, which could be invoked by running C<openssl>
+with no further arguments, was removed in OpenSSL 3.0, and running
+that program with not arguments is now equivalent to C<openssl help>.
```
Comment (t8m) [inline doc/man1/openssl.pod:1412]: s/with not/without/

Comment (paulidale) [reply doc/man1/openssl.pod:1412]: or 'with no'

Comment (levitte) [reply doc/man1/openssl.pod:1412]: "with no" was my intention...

Comment (levitte) [reply doc/man1/openssl.pod:1412]: (I blame gremlins in my keyboard)

File: CHANGES.md [hunk 2]
```diff
@@ -23,6 +23,11 @@ OpenSSL 3.0
 
 ### Changes between 1.1.1 and 3.0 [xx XXX xxxx]
 
+ * Dropped interactive mode from the 'openssl' program.  From now on,
+   the `openssl` command alone is equivalent to `openssl help`.
```
Comment (paulidale) [inline CHANGES.md:27]: s/alone/without arguments/

File: NEWS.md [hunk 3]
```diff
@@ -20,6 +20,7 @@ OpenSSL 3.0
 
 ### Major changes between OpenSSL 1.1.1 and OpenSSL 3.0 [under development]
 
+  * Interactive mode is dropped from the 'openssl' program.
```
Comment (paulidale) [inline NEWS.md:23]: s/dropped/removed/

File: CHANGES.md (hunk 2 above)
Comment (levitte) [reply CHANGES.md]: Fixed

File: NEWS.md (hunk 3 above)
Comment (levitte) [reply NEWS.md]: Fixed

**Comment by iamamoose** [comment]
It's mentioned in the Wiki, but not in detail.   Would also close #4679 and #6292

**Comment by levitte** [comment]
> Would also close #4679 and #6292

Ah, I'll add that in the commit message

**Comment by levitte** [comment]
> There is still the trivial typo change needed.

Er, which one?

**Comment by richsalz** [comment]
Yeah, #4679 is real the "cause of death" for removing interactive mode.

It died alone, unloved, and unmourned.  But I'll raise a glass for the undocumented readline support :)

**Comment by levitte** [comment]
Argh, there were a few spaces in CHANGES.md.  I assume that all approvals stand as long as Travis doesn't get upset again...

**Comment by t8m** [comment]
> Argh, there were a few spaces in CHANGES.md. I assume that all approvals stand as long as Travis doesn't get upset again...

Sure.

**Comment by levitte** [comment]
The vote has reached conclusion: https://mta.openssl.org/pipermail/openssl-project/2020-June/002031.html

Merged.

eca471391378139f76a7d1229b6a5a1dcc4b5603 APPS: Drop interactive mode in the 'openssl' program

**Pull request closed without merging**